src/shareddata.cpp

```diff
@@ -184,9 +184,16 @@ SharedData::initialize(const char *tmpDir,
   }
 
   size_t size = CEIL(SHM_MAX_SIZE, Util::pageSize());
-  void *addr = mmap((void *)sharedDataHeader, size,
-                    PROT_READ | PROT_WRITE, MAP_SHARED,
-                    PROTECTED_SHM_FD, 0);
+  void *addr;
+  if (sharedDataHeader == NULL) {
+    addr = mmap((void *)sharedDataHeader, size,
+                PROT_READ | PROT_WRITE, MAP_SHARED,
+                PROTECTED_SHM_FD, 0);
+  } else {
+    addr = mmap((void *)sharedDataHeader, size,
+                PROT_READ | PROT_WRITE, MAP_SHARED | MAP_FIXED,
```

Comment:
I thought we agreed that we would use map_fixed_noreplace().  The problem with `MAP_FIXED` is that it can silently overwrite a page, and that is really hard to debug.

There is a `map_fixed_noreplace` in: `src/mtcp/mtcp_util.c`
You could always copy it to somewhere else, and use that.  Then a modern Linux will use MAP_FIXED_NOREPLACE and we'll see the bug, and an ancient Linux doesn't matter so much.